test/metabase/lib/field_test.cljc

```diff
@@ -684,3 +688,34 @@
              (lib.metadata.calculation/metadata
               query
               [:field {:lib/uuid "aa0e13af-29b3-4c27-a880-a10c33e55a3e", :base-type :type/Text} 4]))))))
+
+(deftest ^:parallel ref-to-joined-column-from-previous-stage-test
+  (let [query (-> (lib/query meta/metadata-provider (meta/table-metadata :venues))
+                  (lib/join (-> (lib/join-clause
+                                 (meta/table-metadata :categories)
+                                 [(lib/=
+                                   (meta/field-metadata :venues :category-id)
+                                   (lib/with-join-alias (meta/field-metadata :categories :id) "Categories"))])
+                                (lib/with-join-fields [(lib/with-join-alias
+                                                        (meta/field-metadata :categories :name)
+                                                        "Categories")])))
+                  lib/append-stage)
+        joined-col (last (lib/breakoutable-columns query))]
+    (is (=? {:lib/type :metadata/field
```

Comment:
You'll have to update this to `:metadata/column` since my PR to finally change that landed today

Reply:
Right, although interestingly, that makes the test fail. There are bugs lurking here...